js/huobi.js

```diff
@@ -4013,6 +4013,20 @@ module.exports = class huobi extends Exchange {
     }
 
     async createSpotOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        /**
+         * @ignore
+         * @method
+         * @name huobi#createSpotOrder
+         * @description create a spot trade order
+         * @see https://huobiapi.github.io/docs/spot/v1/en/#place-a-new-order
+         * @param {string} symbol unified symbol of the market to create an order in
+         * @param {string} type 'market' or 'limit'
+         * @param {string} side 'buy' or 'sell'
+         * @param {float} amount how much of currency you want to trade in units of base currency
+         * @param {float|undefined} price the price at which the order is to be fullfilled, in units of the quote currency, ignored in market orders
+         * @param {object} params extra parameters specific to the huobi api endpoint
+         * @returns {object} an [order structure]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure}
+         */
         await this.loadMarkets ();
         await this.loadAccounts ();
```

Comment:
@samgermain I think the problem is here (line 4033)

```Javascript
 const accountId = await this.fetchAccountIdByType (market['type']);
 ```
To place a margin order, we must send the appropriate account id (it depends on the margin mode and symbol), currently, we're always sending spot's account.

![image](https://user-images.githubusercontent.com/43336371/187855788-d4146eae-89b4-480b-ad2d-873c1238e81e.png)


Can you please change it accordingly?

Reply:
I created this PR to make it a bit easier to do this, https://github.com/ccxt/ccxt/pull/14698
Huobi is the only exchange giving symbol right now, but I imagine other exchanges will provide if we're getting isolated margin accounts